### DISCUSSION TITLE
Change the texture with the uploaded image

### DISCUSSION THREAD

**Discussion by vttmonkey** [google/model-viewer#3579] [Q&A]
Hi, I am Tri, a Designer. 

I am new to this model-viewer, I learn this for a month and I am doing a showcase project but I am stuck on a function.

I want to change the texture with the uploaded image (just for the demo so I don't need to upload it to the server) by click on a button.

This is for my sample: https://test-3d-texture.trivu9.repl.co

The model: [Custom frame rec.glb.txt](https://github.com/google/model-viewer/files/9099599/Custom.frame.rec.glb.txt)
 
The texture I want to upload and replace with: 
![Custom-frame-rec-2](https://user-images.githubusercontent.com/70497133/178667512-625d783f-93ff-4389-b6ec-16279e3bdeb0.png)

And source code (if need):

```
<head>
  <script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
</head>

<body>
  <div style="width: 280px; height: :500px; border: 1px solid black; border-radius:10px;margin:auto;">
    <model-viewer style="width: 100%; height: 400px;"src="Custom frame rec.glb" ar ar-placement="wall" ar-modes="webxr scene-viewer quick-look" camera-controls alt="A 3D model of some wall art"></model-viewer>
    <div style="width: 160px; font-family: Arial, Helvetica, sans-serif; padding:10px; color:white;background-color: black; text-align: center;margin:auto; margin-bottom: 50px; border-radius: 4px;">Change image</div>
  </div>
</body>
```

Thank you for your help.
PS: Before I open this forum (or something) I try to read all the documents I can find but I don't know the code as much so I really need your help.

**Comment by elalish** [comment]
It's hard to help if you don't ask a question 😄. Try following this example: https://modelviewer.dev/examples/scenegraph/#createTexturesExample

**Reply by vttmonkey** [reply]
Thank you for reply, so sorry I miss that one, I need the java code for button to get (upload) the image from computer and replace to the model.

I already try some but it is not work after all.

**Reply by vttmonkey** [reply]
I deleted my code but my idea is combine the upload image code from this site https://www.codegrepper.com/code-examples/whatever/change+image+src+on+upload+javascript
with the code to change the texture in this site https://stackoverflow.com/questions/71507201/change-the-texture-of-an-existing-model-in-google-model-viewer-using-an-image-fi

**Comment by vttmonkey** [comment]
Can someone help me?

**Reply by elalish** [reply]
We do this in our editor, but you may have to hunt through the code base a bit to find all the pieces you want: https://github.com/google/model-viewer/blob/master/packages/space-opera/src/components/materials_panel/materials_panel.ts#L726

**Reply by vttmonkey** [reply]
oh this is more clearly, thank you for this I will try again.
